src/NServiceBus.Transport.SQS/DelayedMessagesPump.cs

```diff
@@ -168,14 +169,17 @@ IReadOnlyCollection<SqsReceivedDelayedMessage> PrepareMessages(CancellationToken
                         QueueUrl = delayedDeliveryQueueUrl,
                         MessageAttributes =
                         {
-                            [TransportHeaders.DelaySeconds] = new MessageAttributeValue
+                            [DelaySeconds] = new MessageAttributeValue
                             {
                                 StringValue = remainingDelay.ToString(),
                                 DataType = "String"
                             }
                         }
                     };
 
+                    // for native integration scenario's we copy over all the message attributes so we don't lose part of the message when moving to the delayed delivery queue
+                    preparedMessage.CopyMessageAttributes(receivedMessage.MessageAttributes);
```

Comment:
This seems to be done unconditionally, is it safe to do on regular messages?

Reply:
I do think we always have to do it and we should probably slightly tweak the text. Because in any case we want to make sure when we fetch `*` we also want to copy all the attributes every that are necessary. Thoughts @lailabougria

Reply:
Yes exactly, especially due to the fact that we create a mix of attributes on the native message attributes and some in the header dictionary once a message goes through retries. The first time we'd be handling a native message, the second time, a transportmessage.